### PR TITLE
Add support for amp-dailymotion

### PR DESCRIPTION
**Pull request by amedina**
Add amp-dailymotion support.
cc: @mjangda, @pietergreyling, @pbakaus

### CODE REVIEW

File: includes/embeds/class-amp-dailymotion-embed.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
+<?php
+
+require_once( AMP__DIR__ . '/includes/embeds/class-amp-base-embed-handler.php' );
+
+// Much of this class is borrowed from Jetpack embeds
+class AMP_DailyMotion_Embed_Handler extends AMP_Base_Embed_Handler {
+
+	const URL_PATTERN = '#https?:\/\/(www\.)?dailymotion\.com\/video\/.*#i';
+	const RATIO = 0.5625;
```
Comment (dandv) [inline includes/embeds/class-amp-dailymotion-embed.php]: Commend on that magic number?

File: tests/test-amp-dailymotion-embed.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			),
+
+			'url_with_title' => array(
+				'http://www.dailymotion.com/video/x5awwth_snatched-official-trailer-2-hd_shortfilms' . PHP_EOL,
+				'<p><amp-dailymotion data-videoid="x5awwth" layout="responsive" width="600" height="338"></amp-dailymotion></p>' . PHP_EOL
+			),
+
+			'shortcode_unnamed_attr_as_url' => array(
+				'[dailymotion https://www.dailymotion.com/video/x5awwth]' . PHP_EOL,
+                '<amp-dailymotion data-videoid="x5awwth" layout="responsive" width="600" height="338"></amp-dailymotion>' . PHP_EOL
```
Comment (mjangda) [inline tests/test-amp-dailymotion-embed.php]: Minor: fix indentation here.

File: includes/embeds/class-amp-dailymotion-embed.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		if ( empty( $url ) ) {
+			return '';
+		}
+
+		$video_id = $this->get_video_id_from_url( $url );
+
+		return $this->render( array(
+			'url' => $url,
+			'video_id' => $video_id,
+		) );
```
Comment (mjangda) [inline includes/embeds/class-amp-dailymotion-embed.php]: It looks like the shortcode for Dailymotion expects the `id` not the URL, so we may need to modify some of the handling here.

The `oembed` method should call `get_video_id_from_url` to extract the `video_id` and then call `render` directly.

The `shortcode` method can mostly stay as-is, expect we can change references from `$url` => `$video_id` and remove the call to `get_video_id_from_url`. We also need to add a branch to the first `if` block to look for `$attr['id']` and use that if set.

The tests should be updated accordingly as well.

Comment (amedina) [inline includes/embeds/class-amp-dailymotion-embed.php]: Makes sense. Updated the pieces appropriately. The shortcode tests pass, and they work on the AMP pages, but it seems there is an issue with the canonical ([dailymotion id=<video_id>] is not being rendered)

File: includes/embeds/class-amp-dailymotion-embed.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+			)
+		);
+	}
+
+	private function get_video_id_from_url( $url ) {
+		$parsed_url = parse_url( $url );
+		parse_str( $parsed_url['path'], $path );
+        $tok = explode("/", $parsed_url['path']);
+		$tok = explode("_", $tok[2]);
+        $video_id = $tok[0];
```
Comment (amedina) [inline includes/embeds/class-amp-dailymotion-embed.php]: Fixed.

File: tests/test-amp-dailymotion-embed.php (hunk 2 above)
Comment (amedina) [inline tests/test-amp-dailymotion-embed.php]: Fixed.

File: includes/embeds/class-amp-dailymotion-embed.php (hunk 1 above)
Comment (amedina) [inline includes/embeds/class-amp-dailymotion-embed.php]: Makes sense. Will update.

**Pull request merged**